inference-engine/src/transformations/src/transformations/smart_reshape/strided_slice_squeeze.cpp

```diff
@@ -0,0 +1,188 @@
+                opset5::Constant::create(element::i64, {end_vec.size()}, end_vec),
+                opset5::Constant::create(element::i64, {strides_vec.size()}, strides_vec),
+                begin_mask, end_mask, new_axis_mask, shrink_axis_mask, ellipsis_mask);
+
+        replace_node(slice, new_slice);
+        new_slice->set_friendly_name(slice->get_friendly_name());
+        copy_runtime_info(slice, new_slice);
+        return true;
+    };
+    auto m = std::make_shared<ngraph::pattern::Matcher>(ss_label, "SqueezeStridedSlice");
+    register_matcher(m, callback);
+}
+
+NGRAPH_RTTI_DEFINITION(ngraph::pass::SharedSqueeze, "SharedSqueeze", 0);
+
+bool squeezes_perform_the_same(std::shared_ptr<ngraph::opset1::Squeeze> lhs, std::shared_ptr<ngraph::opset1::Squeeze> rhs) {
+    size_t l_input_size = lhs->inputs().size(), r_input_size = rhs->inputs().size();
+    if (l_input_size != r_input_size)
+        return false;
+    if (lhs->inputs().size() == 1 && rhs->inputs().size() == 1)
+        return true;
+    const auto rank = lhs->get_input_partial_shape(0).rank();
+    if (rank.is_dynamic())
+        return false;
+    const auto l_axes = std::dynamic_pointer_cast<ngraph::opset1::Constant>(lhs->get_input_node_shared_ptr(1));
+    const auto r_axes = std::dynamic_pointer_cast<ngraph::opset1::Constant>(rhs->get_input_node_shared_ptr(1));
+    if (l_axes && r_axes)
+        return normalize_axes(lhs->description(), l_axes->cast_vector<int64_t>(), rank) ==
+        normalize_axes(rhs->description(), r_axes->cast_vector<int64_t>(), rank);
+    return false;
+}
+
+bool ngraph::pass::SharedSqueeze::run_on_function(std::shared_ptr<ngraph::Function> f) {
+    OV_ITT_SCOPED_TASK(itt::domains::IETransform, "ngraph::pass::SharedSqueeze");
+
+    bool graph_rewritten = false;
+
+    std::map<ngraph::Output<Node>, std::vector<std::shared_ptr<ngraph::opset1::Squeeze>>> source_to_squeeze;
+    for (const auto & node : f->get_ordered_ops()) {
+        // Recursively apply transformation for sub-graph based operations
+        if (auto sub_graph_node = std::dynamic_pointer_cast<op::util::SubGraphOp>(node)) {
+            if (auto sub_graph = sub_graph_node->get_function()) {
+                graph_rewritten |= run_on_function(sub_graph);
+            }
+        }
+        if (auto squeeze = std::dynamic_pointer_cast<ngraph::opset1::Squeeze>(node)) {
+            source_to_squeeze[squeeze->input_value(0)].push_back(squeeze);
+        }
+    }
+
+    for (auto& pair : source_to_squeeze) {
+        if (pair.second.size() < 2)
+            continue;
+        auto root_squeeze = pair.second[0];
+        for (auto& child_squeeze : pair.second) {
+            if (root_squeeze->get_instance_id() != child_squeeze->get_instance_id() && squeezes_perform_the_same(root_squeeze, child_squeeze)) {
+                graph_rewritten |= replace_output_update_name(child_squeeze->output(0), root_squeeze->output(0));
```

Comment:
That could lead to some issues with output names for cases when root Squeeze has a Result operation or more than two Squeeze operations have a Result operation. But I hope we will resolve naming issue faster than we will get such model.